web/server/api/endpoints/files.py

```diff
@@ -72,72 +148,32 @@ def walk_path(
                     )
         return sorted(directories, key=lambda x: x.name), sorted(files, key=lambda x: x.name)
 
-    directories, files = walk_path(settings.project_path)
+    directories, files = walk_path(path)
+    relative_path = str(Path(path).relative_to(settings.project_path))
     return models.Directory(
-        name=os.path.basename(settings.project_path),
-        path="",
+        name=os.path.basename(path),
+        path="" if relative_path == "." else relative_path,
         directories=directories,
         files=files,
     )
 
 
-@router.get("/{path:path}", response_model=models.File)
-def get_file(
-    path: str = Depends(validate_path),
-    settings: Settings = Depends(get_settings),
-    path_mapping: t.Dict[Path, models.FileType] = Depends(get_path_mapping),
+def _get_file_with_content(
+    path: Path,
+    settings: Settings,
+    path_mapping: t.Dict[Path, models.FileType],
 ) -> models.File:
     """Get a file, including its contents."""
+    file_path = settings.project_path / path
+
     try:
-        with open(settings.project_path / path) as f:
+        with open(file_path) as f:
             content = f.read()
-    except FileNotFoundError:
-        raise HTTPException(status_code=HTTP_404_NOT_FOUND)
+    except FileNotFoundError as e:
+        raise e
```

Comment:
Do you want to handle or log the exception? If not, no need for the try/catch block.